### ISSUE TITLE
Add proof irrelevance proofs for common relations

### ISSUE THREAD

**Issue by gallais** [agda/agda-stdlib#176]
The shape of a proof `_≤_` on `ℕ` is for instance completely determined by the two natural numbers involved. I am sure there are plenty of similar low hanging fruits.

Not sure whether we should introduce a notion of `Proof-Irrelevance` just like we have e.g. `Associative` and refactor [`proof-irrelevance`](https://agda.github.io/agda-stdlib/Relation.Binary.PropositionalEquality.html#1420)'s type. After all, it's not a property of a binary relation per se but rather of the Set. I supposed we could alternatively call this notion `isProp` after the [mere proposition](https://ncatlab.org/nlab/show/mere+proposition) vocabulary in hott.

**Comment by MatthewDaggitt** [comment]
Yes, I have the irrelevance of `_≤_` in my own library as well. It's often necessary to prove congruences for functions using the `WellFounded` machinery.

I like the idea of a general notion. The type of the property over `Rel` is obvious to me, but what would the general form over `Set` look like? Drawing a blank at the moment...

**Comment by gallais** [comment]
I do think that `isProp` is the right notion here. We could have:

```agda
open import Level
open import Relation.Unary
open import Relation.Binary
open import Agda.Builtin.Equality

isProp : ∀ {a} → Set a → Set a
isProp A = (a b : A) → a ≡ b

Irrelevant-Pred : ∀ {a ℓ} {A : Set a} → Pred A ℓ → Set (ℓ ⊔ a)
Irrelevant-Pred P = ∀ {x} → isProp (P x)

Irrelevant-Rel : ∀ {a ℓ} {A : Set a} → Rel A ℓ → Set (ℓ ⊔ a)
Irrelevant-Rel R = ∀ {x y} → isProp (R x y)
```

**Comment by MatthewDaggitt** [comment]
Closed with #189
